### PR TITLE
Add versioning release instructions, add boto3 version range

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*
- Add versioning release instructions
- add boto3 version range
- add requests version range
- add inline comments for where version ranges are defined

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by sxjscience — APPROVED** [review]
LGTM!

File: docs/ReleaseInstructions.md [hunk 1]
```diff
@@ -3,6 +3,35 @@
 ### Prior to release: 1 week out
 
 * Ensure the version specified in `docs/config.ini`, `VERSION`, `docs/index.rst`, and `docs/badges.rst` align with the intended release version.
+* Check all dependency version ranges.
+  * Ensure all dependencies are not capped by major version, unless the reason is documented inline.
+    * Example of major version cap: `scikit-learn<2`
+  * Ensure all dependencies have an upper cap, unless the reason is documented inline.
+    * Example of no upper cap: `scikit-learn>=1.0`
```
Comment (gradientsky) [inline docs/ReleaseInstructions.md:10]: Discussion point: an Introduction of a overly aggressive upper cap would require binary releases every time we want to change the range. This significantly increase work for DLC patching (~1 wk becomes 3-4 weeks).
Pinning pros: repeatable builds
Pinning cons: require more frequent releases as the new CVEs coming.

File: docs/ReleaseInstructions.md [hunk 2]
```diff
@@ -3,6 +3,35 @@
...
+  * Note: For micro releases such as AutoGluon 0.6.2, this is optional.
+  * If increasing the range causes an error, either:
+    1. Fix the error.
+    2. Avoid the range change and add an inline comment in setup.py that an error occurred and why we didn't fix.
+  * If increasing the range causes a warning, either:
+    1. Fix the warning.
+    2. Suppress the warning for the user + provide justification and appropriate TODOs.
+    3. Avoid the range change and add an inline comment in setup.py that a warning occurred and why we didn't fix.
+  * Ensure CI passes, potentially benchmark to catch performance regressions / more complex bugs.
+  * Note: To truly catch potential errors, you will need to use the latest supported Python version, since some packages may only support the newer Python version in their latest releases.
```
Comment (gradientsky) [inline docs/ReleaseInstructions.md:34]: We really need to test this on both sides of supported ranges: i.e. numpy at some point stopped 3.7 support; so 3.7 was stuck at <1.21, while 3.8/3.9 was 1.21+

File: timeseries/setup.py [hunk 3]
```diff
@@ -24,16 +24,16 @@
...
     # version ranges added in ag.get_dependency_version_ranges()
     "joblib~=1.1",
-    "numpy",
-    "scipy",
-    "pandas",
+    "numpy",  # version range defined in `core/_setup_utils.py`
+    "scipy",  # version range defined in `core/_setup_utils.py`
+    "pandas",  # version range defined in `core/_setup_utils.py`
     "statsmodels~=0.13.0",
     "gluonts~=0.12.0",
```
Comment (gradientsky) [inline timeseries/setup.py:31]: `~=0.1.3.0` -> `~=0.13` ?

File: timeseries/setup.py [hunk 4]
```diff
@@ -24,16 +24,16 @@
...
     "torch>=1.9,<1.14",
     "pytorch-lightning>=1.7.4,<1.9.0",
-    "networkx",
+    "networkx",  # version range defined in `core/_setup_utils.py`
     "statsforecast==1.4.0",
```
Comment (gradientsky) [inline timeseries/setup.py:35]: ditto; or if we pin, needs a comment

File: docs/ReleaseInstructions.md (hunk 2 above)
Comment (Innixma) [reply docs/ReleaseInstructions.md:34]: Yes, technically it is more complex to comprehensively test. We can add additional steps in future PRs.

File: docs/ReleaseInstructions.md (hunk 1 above)
Comment (Innixma) [reply docs/ReleaseInstructions.md:10]: We already have aggressive upper caps on almost every dependency. We are planning to relax some of them this release (Numpy and SciPy)

File: timeseries/setup.py (hunk 3 above)
Comment (Innixma) [reply timeseries/setup.py:31]: This PR is not touching the version ranges to minimize potential merge conflicts with other PRs. A follow-up PR will be sent to clean up the ranges.

File: timeseries/setup.py (hunk 4 above)
Comment (Innixma) [reply timeseries/setup.py:35]: Tracked in #2831

**Review by gradientsky — APPROVED** [review]

**Pull request merged**